### PR TITLE
feat(threaded-replies): Implement Create Replies and Replies Toggle

### PR DESCRIPTION
**Pull request by JChan106**
- [x] Unit tests
- [x] Screenshots/ gifs

HasReplies enabled and hasNewThreadedReplies enabled:
image
Show Replies:

Create Replies:

HasReplies disabled and hasNewThreadedReplies enabled:
Resolve/Unresolve:

hasReplies enabled and hasNewThreadedReplies disabled:
Create and Show Replies:

### CODE REVIEW

File: src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js [hunk 1]
```diff
@@ -183,13 +183,17 @@ const ActiveState = ({
                                         mentionSelectorContacts={mentionSelectorContacts}
                                         onDelete={onCommentDelete}
                                         onEdit={onCommentEdit}
+                                        onReplyCreate={reply => onReplyCreate(item.id, FEED_ITEM_TYPE_COMMENT, reply)}
```
[the remark this answers is not in the document]
Comment (karelee7) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js:186]: Also, I see we're using `item.id` multiple times; should we destructure it above? (i.e. `const { id } = item;`)

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js [hunk 2]
```diff
@@ -324,8 +324,13 @@ const BaseComment = (props: BaseCommentProps) => {
                 <Replies
                     {...commentProps}
                     isRepliesLoading={isRepliesLoading}
-                    onReplySelect={onReplySelect}
+                    isParentPending={isPending}
+                    onReplySelect={onSelect}
+                    onReplyCreate={onReplyCreate}
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/BaseComment.js:329]: minor: alphabetical order!

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js [hunk 3]
```diff
@@ -365,8 +381,29 @@ const Replies = ({
...
+    const handleSubmitNewReply = (reply: string) => {
+        setShowReplyForm(false);
+        onReplyCreate(reply);
+    };
+
     return (
         <div className="bcs-Replies">
+            <RepliesToggle
+                onShowReplies={onShowReplies}
+                onHideReplies={index => onHideReplies([replies[index]])}
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/BaseComment.js:403]: minor: alphabetical order!

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js [hunk 4]
```diff
@@ -397,6 +435,16 @@ const Replies = ({
                     })}
                 </ol>
             </div>
+            <CreateReply
+                mentionSelectorContacts={mentionSelectorContacts}
+                getMentionWithQuery={getMentionWithQuery}
+                isDisabled={isParentPending}
+                onFocus={() => onReplySelect(true)}
+                onCancel={handleCancelNewReply}
```
Comment (karelee7) [inline src/elements/content-sidebar/activity-feed/comment/BaseComment.js:443]: minor: alphabetical order!

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js [hunk 5]
```diff
@@ -350,12 +360,18 @@ const Replies = ({
     getAvatarUrl,
     getMentionWithQuery,
     getUserProfileUrl,
+    isParentPending = false,
     isRepliesLoading = false,
     mentionSelectorContacts,
+    onReplyCreate = noop,
```
Comment (bfoxx1906) [inline src/elements/content-sidebar/activity-feed/comment/BaseComment.js:366]: Will this function even make sense if the is a noop operation? Should we make onReplyCreate non optional?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js:186]: Those are the handlers from the legacy code we will be removing. A lot of these cbs don't need handlers imo if there is n't steps to the logic. In this case, its just adding another parameter.

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js (hunk 5 above)
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/comment/BaseComment.js:366]: What if we don't want replies for a comment? They shouldn't need to add a cb for this then right?

Comment (bfoxx1906) [reply src/elements/content-sidebar/activity-feed/comment/BaseComment.js:366]: If we don't want to show replies then why even render the component?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js:186]: And yeah I've tried that in the past, but it didn't work for some reason, Ill revisit

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js (hunk 5 above)
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/comment/BaseComment.js:366]: What do you mean? Shouldn't we have flexibility in having a BaseComment not have replies? Are you saying that all comments need to have the ability to create replies? We are also reusing BaseComment in the `Replies` component. That would mean we have to pass the create prop in each reply as well.

File: src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JChan106) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js]: looks like the linter doesn't allow it, will keep as is.

**Review by karelee7 — APPROVED** [review]
Looks good to me!

**Review by bfoxx1906 — APPROVED** [review]

**Review by patlm — APPROVED** [review]

**Pull request merged**